File: AdventOfCode2022/Day01/Day01.hs
import Data.List.Split (splitWhen)
import Data.List (sort)

toInt x = read x :: Integer
sum' = sum . map toInt

main = do
  content <- readFile "input.txt"
  let calories = reverse . sort . map sum' . splitWhen (=="") $ lines content
  
  -- part 1
  print $ head calories

  -- part 2
  print $ sum . take 3 $ calories

File: AdventOfCode2022/Day01/day01.js
const fs = require("fs");

const data = fs.readFileSync("Day01/input.txt", "utf8");

const sums = [];
const lines = data.split("\n");
let max = 0;
let sum = 0;
for (let i = 0; i < lines.length; i++) {
  let line = lines[i].trim();
  if (!line) {
    if (sum > max) {
      max = sum;
    }
    sums.push(sum);
    sum = 0;
  } else {
    sum += +line;
  }
}

// Part 1
console.log(max);

// Part 2
sums.sort();
sums.reverse();
console.log(sums[0] + sums[1] + sums[2]);
